### PR TITLE
VS2015: fix possible unreached code warning in VS2015

### PR DESCRIPTION
**Pull request by vlserov**
### Description 
Warning possible unreached code in VS2015.

Fixes # - _issue number(s) if exists_

- [x] - git commit message contains appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneTBB/blob/master/CONTRIBUTING.md#pull-requests) for details)_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add respective label(s) to PR if you have permissions_

- [ ] bug fix - _change which fixes an issue_
- [ ] new feature - _change which adds functionality_
- [x] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and for some bug fixes_
- [x] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: test/tbb/test_limiter_node.cpp [hunk 1]
```diff
@@ -623,5 +623,5 @@ TEST_CASE("Test correct node deallocation while using small_object_pool") {
     graph.wait_for_all();
 
     tbb::task_scheduler_handle handle = tbb::task_scheduler_handle::get();
-    REQUIRE_NOTHROW( tbb::finalize( handle, std::nothrow ) );
```
Comment (anton-potapov) [inline test/tbb/test_limiter_node.cpp:626]: @vlserov could you please clarify that the compiler complains about, and why this change fixes the problem? As, at least to me, changes seems a bit of irrational :)

Comment (vlserov) [reply test/tbb/test_limiter_node.cpp:626]: Since `tbb::finalize` is marked `noexcept`, MSVC 2015 compiler assumes that some part of macro `REQUIRE_NOTHROW` cannot be reached. All tests are assembled with `/EHsc`(Standard C++ exception handling model), Structured Exceptions (SEH) cannot be handled by C++ `try..catch` block without `/EHa` flag.
`REQUIRE_NOTHROW` can be removed.
Another possible solution is using `/EHa` flag for test_limiter_node.

**Review by alexey-katranov — APPROVED** [review]

**Review by alexey-katranov — APPROVED** [review]
